storagedriver/azure/randomwriter.go

```diff
@@ -0,0 +1,208 @@
+package azure
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	azure "github.com/MSOpenTech/azure-sdk-for-go/clients/storage"
+)
+
+// blockStorage is the interface required from a block storage service
+// client implementation
+type blockStorage interface {
+	CreateBlockBlob(container, blob string) error
+	GetBlob(container, blob string) (io.ReadCloser, error)
+	GetBlobRange(container, blob string, start, length int64) (io.ReadCloser, error)
+	PutBlock(container, blob, blockID string, chunk []byte) error
+	GetBlockList(container, blob string, blockType azure.BlockListType) (azure.BlockListResponse, error)
+	PutBlockList(container, blob string, blocks []azure.Block) error
+}
+
+// randomBlobWriter enables random access semantics on Azure block blobs
+// by enabling writing arbitrary length of chunks to arbitrary write offsets
+// within the blob. Normally, Azure Blob Storage does not support random
+// access semantics on block blobs; however, this writer can download, split and
+// reupload the overlapping blocks and discards those being overwritten entirely.
+type randomBlobWriter struct {
+	bs        blockStorage
+	blockSize int
+}
+
+func newRandomBlobWriter(bs blockStorage, blockSize int) randomBlobWriter {
+	return randomBlobWriter{bs: bs, blockSize: blockSize}
+}
+
+// Write writes the given chunk to the specified position of an existing blob.
+// The offset must be equals to size of the blob or smaller than it.
+func (r *randomBlobWriter) Write(container, blob string, offset int64, chunk io.Reader) (int64, error) {
```

Comment:
True, but it is exported on the type.  Is it part of an interface? It may be better to call it `WriteBlobAt` or something like that. I think having it called `Write` is tripping me up.